### PR TITLE
fixes unscope where with a grouped right-hand side

### LINKED ISSUE #30967: ActiveRecord does not unscope where conditions when right-hand side is a group
### Steps to reproduce
https://gist.github.com/jemminger/b14dc10bc1501c9efce081729182de33

### Expected behavior
Calling `unscope` for `:where` on a column with a group RH side should remove that column's conditions

### Actual behavior
The column's conditions remain

### System configuration
**Rails version**: 5.1.3

**Ruby version**: 2.3.4

### PR DESCRIPTION
**Pull request by jemminger**
### Summary

Fixes #30967

### CODE REVIEW

File: activerecord/lib/active_record/relation/where_clause.rb [hunk 1]
```diff
@@ -135,14 +135,37 @@ def except_predicates_and_binds(columns)
           except_binds = []
           binds_index = 0
 
+          node_tester = -> (node, columns) {
+            binds_contains = node.grep(Arel::Nodes::BindParam).size
+            subrelation = (node.left.kind_of?(Arel::Attributes::Attribute) ? node.left : node.right)
+            included = columns.include?(subrelation.name.to_s)
+            [included, binds_contains]
+          }
+          relevant_nodes = [Arel::Nodes::Between, Arel::Nodes::In, Arel::Nodes::NotIn, Arel::Nodes::Equality, Arel::Nodes::NotEqual, Arel::Nodes::LessThan, Arel::Nodes::LessThanOrEqual, Arel::Nodes::GreaterThan, Arel::Nodes::GreaterThanOrEqual]
```
Comment (gmcgibbon) [inline activerecord/lib/active_record/relation/where_clause.rb]: I think this should be a constant.

**Review by gmcgibbon — COMMENTED** [review]
Please add tests and a changelog entry. You should also be submitting PRs to master.

**Comment by jemminger** [comment]
@gmcgibbon 

> Please add tests and a changelog entry. 

Thanks, will do

> You should also be submitting PRs to master.

I also submitted a patch for 5-2-stable (https://github.com/rails/rails/pull/34675), should I just make this change in master instead and close these two PRs?

**Comment by gmcgibbon** [comment]
We can backport the original commit if it looks the same. You should be able to [change the base branch](https://help.github.com/articles/changing-the-base-branch-of-a-pull-request/) of this PR to master.

**Comment by jemminger** [comment]
OK - the relevant file https://github.com/rails/rails/blob/master/activerecord/lib/active_record/relation/where_clause.rb changes from 5.1 to 5.2, so my 5-2-stable patch (https://github.com/rails/rails/pull/34675) is closer.  

I changed its base branch to master, but now it says 562 changed files instead of my 3, is that expected?

Sorry for all the questions, first time trying to submit a patch.

**Comment by jemminger** [comment]
On second thought, that's probably because I built the patch off of 5-2-stable.  I should do it over based on master, correct?

**Comment by jemminger** [comment]
Built patch against master, new PR at #34679

**Pull request closed without merging**